gen/enum_test.go

```diff
@@ -399,3 +399,27 @@ func TestEnumAccessors(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumWireLabel(t *testing.T) {
+	tests := []struct {
+		enumItem  te.EnumWithWireLabel
+		wireValue string
+	}{
+		{te.EnumWithWireLabelUsername, "surname"},
+		{te.EnumWithWireLabelPassword, "hashed_password"},
+		{te.EnumWithWireLabelSalt, "salt"},
+	}
+	for _, tt := range tests {
+		t.Run("compare label:"+tt.wireValue, func(t *testing.T) {
+			label := te.EnumWithWireLabel(tt.enumItem)
+			b, err := label.MarshalText()
+			assert.NoError(t, err)
+			assert.Equal(t, tt.wireValue, string(b))
+			assertRoundTrip(
+				t, &tt.enumItem,
+				wire.NewValueI32(int32(tt.enumItem)),
+				"test roundtrip "+tt.wireValue,
+			)
+		})
+	}
```

Comment:
Can we also confirm through an integration test that this effects the desired behavior for JSON marshaling?

Can we also test the behavior of serializing an unrecognized enum case?

Reply:
👍

Reply:
just want to make sure I understand your comment correctly, are you suggesting something here like:
```go
...
    // marshal
    label := te.EnumWithWireLabel(te.EnumWithWireLabelUsername)
    b, err := json.Marshal(label)
    assert.Equal(t, "username", string(b))

    //  unmarlshal      
    var testLabel te.EnumWithWireLabel
    err = json.Unmarshal(b, testLabel)
    assert.Equal(t, testLabel, label)
... 
```

or you're suggesting a specific kind of integration tests? Thanks

Reply:
This is fine.